src/st/strategy/shortcode/class-wpml-pb-update-shortcodes-in-content.php

```diff
@@ -94,13 +95,30 @@ private function replace_string_with_translation( $block, $original, $translatio
 					$pattern     = '/' . $attr . '=(["\'])' . preg_quote( $original, '/' ) . '(["\'])/';
 					$replacement = $attr . '=${1}' . $this->escape_backward_reference_on_replacement_string( $translation ) . '${2}';
 				} else {
+					$has_wrapper = false !== strpos( $new_block, '[' . WPML_PB_Shortcode_Content_Wrapper::WRAPPER_SHORTCODE_NAME . ']' );
 					$pattern     = '/(]\s*)' . preg_quote( trim( $original ), '/' ) . '(\s*\[)/';
 					$replacement = '${1}' . $this->escape_backward_reference_on_replacement_string( trim( $translation ) ) . '${2}';
 				}
 
-				$new_block         = preg_replace( $pattern, $replacement, $block );
-				$replacement       = $this->escape_backward_reference_on_replacement_string( $new_block );
-				$this->new_content = preg_replace( '/' . preg_quote( $block, '/' ) . '/', $replacement, $this->new_content, 1 );
+				$new_block = preg_replace( $pattern, $replacement, $block );
```

Comment:
@strategio This code seems to complicated to me. Why can we just run code to remove the wrappers by default?

```
$new_block         = preg_replace( $pattern, $replacement, $block );
$new_block         = $this->remove_wrappers( $new_block );
$replacement       = $this->escape_backward_reference_on_replacement_string( $new_block );
$this->new_content = preg_replace( '/' . preg_quote( $block, '/' ) . '/', $replacement, $this->new_content, 1 );
```

There's also a bug in the code where the code remove wrappers doesn't run if `is_string_too_long_for_regex` return true.